src/containers/my-resources/add-resource-with-input/AddResourceWithInput.tsx

```diff
@@ -0,0 +1,63 @@
+import { useState, ChangeEvent, FC, MutableRefObject } from 'react'
+import { useTranslation } from 'react-i18next'
+import SearchIcon from '@mui/icons-material/Search'
+import Box from '@mui/material/Box'
+
+import { useDebounce } from '~/hooks/use-debounce'
+import AppButton from '~/components/app-button/AppButton'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+
+import { styles } from '~/containers/my-resources/add-resource-with-input/AddResourceWithInput.styles'
+
+interface AddResourceWithInputProps {
+  btnText: string
+  fetchData: () => Promise<void>
+  onClick: () => void
+  searchRef: MutableRefObject<string>
+}
+
+const AddResourceWithInput: FC<AddResourceWithInputProps> = ({
+  btnText,
+  fetchData,
+  onClick,
+  searchRef
+}) => {
+  const { t } = useTranslation()
+  const [searchInput, setSearchInput] = useState<string>('')
+
+  const debounceOnChange = useDebounce((text: string) => {
+    searchRef.current = text
+    void fetchData()
+  })
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(e.target.value)
+    debounceOnChange(e.target.value)
+  }
+
+  const onClear = () => {
+    setSearchInput('')
+    searchRef.current = ''
+    void fetchData()
+  }
+
+  return (
+    <Box sx={styles.container}>
+      <AppButton onClick={onClick} sx={styles.addBtn}>
+        {t(btnText)}
+        <span style={styles.addIcon}>+</span>
```

Comment:
Use Typography with component span and use plus icon from mui

Reply:
fixed